controller/kubernetes_pod_controller.go

```diff
@@ -171,6 +178,121 @@ func (kc *KubernetesPodController) syncHandler(key string) (err error) {
 	return nil
 }
 
+// handleWorkloadPodDeletionIfCSIPluginPodIsDown deletes work load pod on the same node
+// of the CSI plugin pod if the CSI plugin pod is down.
+// This is required if the cluster is using storage network only. When the cluster
+// is using the cluster network, the CSI plugin pod is using host PID namespace hence the mount
+// entry exists on the host namespace which retains after the CSI plugin pod is down.
+// When the storage network is set up, the CSI plugin pod uses the container PID namespace because
+// Multus network only exist in the Kubernetes network and relies on the Kubernetes DNS for
+// the mount point, hence the lifecycle of the mount entry goes with the container. Hence when
+// the CSI plugin pod is down, Longhorn need to delete the share manager pod to trigger the
+// remount
+func (kc *KubernetesPodController) handleWorkloadPodDeletionIfCSIPluginPodIsDown(csiPod *corev1.Pod) error {
+	if csiPod.DeletionTimestamp.IsZero() {
```

Comment:
I am not sure which scenario the controller would miss the DeletionTimestamp. The object shouldn't be replaced since the new CSI pod would have a different name. Are you suggesting that the csiPod object would be nil in this case?

Reply:
> Perhaps we can do something similar here (e.g. filter out pods created AFTER the CSI plugin pod's deletion timestamp)?

Good idea. Updated, thank you!